numpy/core/src/multiarray/item_selection.c

```diff
@@ -2202,7 +2206,164 @@ count_nonzero_bytes(const npy_uint8 *d, npy_uintp unrollx)
     return unrollx - zero_count;
 }
 
+
+static NPY_INLINE NPY_GCC_OPT_3 npy_uintp
+count_nonzero_int16_simd(npy_int16 *d, npy_uintp unrollx)
+{
+    npy_uintp zero_count = 0;
+    const npy_uintp innerloop_jump = NPY_MAX_UINT16;
+    const npy_int16 *end = d + unrollx;
+
+    const npyv_u16 vone = npyv_setall_u16(1); 
+    const npyv_u16 vzero = npyv_zero_u16();   
+
+    npy_int16 *target = d;
+    while (d<end) {
+        npyv_u16 vsum16 = npyv_zero_u16(); 
+        target = PyArray_MIN(target+innerloop_jump, end);
+        for (; d<target; d+=npyv_nlanes_u16) {
+            npyv_u16 vt = npyv_cvt_u16_b16(npyv_cmpeq_u16(npyv_load_u16(d), vzero)); 
+            vt = npyv_and_u16(vt, vone);
+            vsum16 = npyv_add_u16(vsum16, vt);
+        }
+
+        zero_count += npyv_sumup_u16(vsum16);
+    }
+
+    return unrollx - zero_count;
+}
+
+
+static NPY_INLINE NPY_GCC_OPT_3 npy_uintp
+count_nonzero_int32_simd(npy_int32 *d, npy_uintp unrollx)
+{
+    npy_uintp zero_count = 0;
+    const npy_uintp innerloop_jump = NPY_MAX_UINT32;
+    const npy_int32 *end = d + unrollx;
+
+    const npyv_u32 vone = npyv_setall_u32(1); 
+    const npyv_u32 vzero = npyv_zero_u32();   
+
+    npy_int32 *target = d;
+    while (d<end) {
+        npyv_u32 vsum32 = npyv_zero_u32(); 
+        target = PyArray_MIN(target+innerloop_jump, end);
+        for (; d<target; d+=npyv_nlanes_u32) {
```

Comment:
```suggestion
                for (npy_intp i = 0; d < end && i < NPY_MAX_UINT32; ++i, d += npyv_nlanes_u32) {
```
same as `count_nonzero_int16_simd`